pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -569,6 +569,12 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 	}
 
 	if pipelineRunFacts.State.IsBeforeFirstTaskRun() {
+		if err := resources.ValidateMatrixPipelineParameterTypes(pipelineRunFacts.State); err != nil {
+			logger.Errorf("Failed to validate matrix %q with error %v", pr.Name, err)
+			pr.Status.MarkFailed(ReasonInvalidTaskResultReference, err.Error())
+			return controller.NewPermanentError(err)
+		}
+
```

Comment:
please add a test in pipelinerun_test.go validating the expected behavior 🙏🏾

Reply:
Added a test in `pipelinerun_test.go` verifying that it works. Not sure if there's a way to test the failure..